nodebuilder/node/metrics.go

```
@@ -0,0 +1,62 @@
+package node
+
+import (
+	"context"
+	"sync/atomic"
+	"time"
+
+	logging "github.com/ipfs/go-log/v2"
+
+	"go.opentelemetry.io/otel/metric/global"
+	"go.opentelemetry.io/otel/metric/instrument"
+)
+
+var (
+	meter = global.MeterProvider().Meter("node")
+	log   = logging.Logger("node")
+)
+
+// WithMetrics registers node metrics.
+func WithMetrics() {
```

please return error here instead of doing log.Fatal